arkouda/groupbyclass.py

```diff
@@ -742,7 +763,27 @@ def argmax(self, values: pdarray) -> Tuple[groupable, pdarray]:
         (array([2, 3, 4]), array([9, 3, 2]))
         """
 
-        return self.aggregate(values, "argmax")
+        k, v = self.aggregate(values, "argmax")
+        return k, cast(pdarray, v)
+
+    def _nested_grouping_helper(self, values: groupable) -> groupable:
+        unique_key_idx = self.broadcast(arange(self.ngroups), permute=True)
+        if hasattr(values, "_get_grouping_keys"):
+            # All single-array groupable types must have a _get_grouping_keys method
+            if isinstance(values, pdarray) and cast(pdarray, values).dtype == akfloat64:
+                raise TypeError("grouping/uniquing unsupported for float64 arrays")
```

Comment:
python's `and` does short circuiting, so the second clause is only ever evaluated if the first part is true, so we don't need the explicit cast since we know `values` is an instance of `pdarray`

So we can simplify a bit to just 
```python
if isinstance(values, pdarray) and values.dtype == akfloat64:
```